src/components/modals/PlaceOrderModal.tsx

```diff
@@ -0,0 +1,66 @@
+'use client';
+
+import React from 'react';
+import { Button } from '@/components/ui/button';
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogDescription,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog';
+import { PlaceOrderForm } from '@/components/forms/PlaceOrderForm';
+import { FaArrowRight } from 'react-icons/fa6';
+
+interface PlaceOrderModalProps {
+  isOpen: boolean;
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  brandName: string;
+  modelName: string;
+  selectedColor: ColorOption;
+}
+
+export const PlaceOrderModal: React.FC<PlaceOrderModalProps> = ({
+  isOpen,
+  setIsOpen,
+  brandName,
+  modelName,
+  selectedColor,
+}) => {
+  return (
+    <Dialog open={isOpen} onOpenChange={setIsOpen}>
+      <DialogTrigger asChild>
+        <Button icon={<FaArrowRight />} iconPosition='right'>
+          Order now
+        </Button>
+      </DialogTrigger>
+      <DialogContent className='order-modal h-full w-full md:flex md:gap-0 md:border-0 md:p-0'>
```

Comment:
🟡 The purpose of the  order-modal class isn't entirely clear. This can be achieved with Tailwind CSS, consider replacing it with utility classes.